clients/unshaded/src/main/java/tachyon/client/RemoteBlockInStream.java

```diff
@@ -138,6 +142,23 @@
   }
 
   /**
+   * An alternative to construct a RemoteBlockInstream, bypassing all the checks.
+   * The returned RemoteBlockInStream can be used to call {#link #readRemoteByteBuffer}.
+   * TODO: remove this constructor, and modify all places using this constructor.
+   *
+   * @param file, any, could be null
+   * @param readType, any type
+   * @param blockIndex, any index
+   * @param ufsConf, any ufs configuration
+   * @param tachyonConf, any
+   * @param addFlag, add another field so that this constructor differentiates
+   */
+  public RemoteBlockInStream(TachyonFile file, ReadType readType, int blockIndex, Object ufsConf,
```

Comment:
So, right now, the "fake" constructor is exposed, and once that is created, you just use the RemoteBlockInStream as before.

I was wondering if it would be cleaner to only expose a static method (and not the "fake" constructor), and the static method uses the "fake" constructor (which is now private), and uses it and returns what needs to be returned. If this is possible, then only the public static method would use the "fake" constructor, and nothing else could accidentally use that constructor.